frontend/src/views/Lobby.tsx

```diff
@@ -337,19 +343,64 @@ function LobbyPage() {
       conditionallyBootKickedUser(room, currentUser);
     };
 
+    setLoading(true);
     connect(roomId, userId).then(() => {
       // Body encrypt through JSON.
       subscribe(routes(roomId).subscribe_lobby, subscribeCallback).then((subscriptionParam) => {
         setSubscription(subscriptionParam);
         setSocketConnected(true);
+        setError('');
+        setLoading(false);
       }).catch((err) => {
         setError(err.message);
+        setLoading(false);
       });
     }).catch((err) => {
       setError(err.message);
+      setLoading(false);
     });
   }, [currentUser, conditionallyBootKickedUser]);
 
+  const refreshRoomDetails = () => {
+    // Call GET endpoint to get latest room info
+    if (!loading) {
+      setLoading(true);
+      getRoom(location.state.roomId)
+        .then((res) => {
+          setStateFromRoom(res);
+
+          // Boot the user from the room, if they are not present.
+          updateCurrentUserDetails(res.users);
+
+          // Attempt to connect the user to the socket.
+          if (currentUser && currentUser.userId) {
+            connectUserToRoom(res.roomId, currentUser.userId);
+          }
+        })
+        .catch((err) => setError(err));
+    }
+  };
+
+  const leaveRoom = () => {
+    // eslint-disable-next-line no-alert
+    if (window.confirm('Are you sure you want to leave the room?')) {
+      if (currentUser && currentUser.userId) {
+        setLoading(true);
+        setError('');
+        removeUser(currentRoomId, {
+          initiator: currentUser,
+          userToDelete: currentUser,
+        });
+        disconnect();
+      }
+    }
+
+    // Redirect user regardless of POST request success.
+    history.replace('/game/join', {
+      error: errorHandler('You left the room.'),
+    });
```

Comment:
I think this should be moved into the if statement - there's a slight bug rn where even if you click cancel you'll still be redirect away from the room (although not kicked on the backend)

Reply:
Ah good catch, thanks, I'll update this.

Reply:
Done ✅